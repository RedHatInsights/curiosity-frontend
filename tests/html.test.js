const fs = require('fs');

describe('Index.HTML', () => {
  const outputDir = 'dist';
  const loadFile = file => (fs.existsSync(file) && fs.readFileSync(file, { encoding: 'utf-8' })) || '';

  it('should have a specific html output', () => {
    const fileContents = loadFile(`./${outputDir}/index.html`);
    const replacedFile = fileContents
      .replace(/</g, '\n<')
      .replace(/"([/a-z0-9]*)\/apps\//gi, '"*/apps/')
      .replace(/\/([a-z0-9]+)\.([a-z0-9]+)\.chunk.css"/gi, '/*.chunk.css"')
      .replace(/\/([a-z0-9]+)\.([a-z0-9]+)\.chunk.js"/gi, '/*.chunk.js"')
      .replace(/\/([a-z]+)\.([a-z0-9]+)\.js"/gi, '/$1.*.js"')
      .replace(/<script>([\d\D])+?<\/script>/gi, '<script>/* filtered */</script>');

    expect(replacedFile).toMatchSnapshot('html output');
  });
});
