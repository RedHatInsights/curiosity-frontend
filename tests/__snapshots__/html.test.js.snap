// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index.HTML should have a specific html output: html output 1`] = `
"
<!doctype html>
<html lang=\\"en\\">
<head>
<meta charset=\\"utf-8\\"/>
<title>Subscription Reporting
</title>
<esi:include src=\\"/@@env/chrome/snippets/head.html\\" />
<link href=\\"*/apps/subscriptions/static/css/*.chunk.css\\" rel=\\"stylesheet\\">
<link href=\\"*/apps/subscriptions/static/css/*.chunk.css\\" rel=\\"stylesheet\\">
</head>
<body>
<esi:include src=\\"/@@env/chrome/snippets/body.html\\" />
<script>/* filtered */</script>
<script src=\\"*/apps/subscriptions/static/js/*.chunk.js\\">
</script>
<script src=\\"*/apps/subscriptions/static/js/*.chunk.js\\">
</script>
</body>
</html>"
`;
